### ISSUE TITLE
Be more resilient to nixpkgs errors

### ISSUE THREAD

**Issue by ncfavier** [NixOS/nixos-search#391]
It's not the first time I see the import fail because e.g. someone put a string in `meta.maintainers` (cf. https://github.com/NixOS/nixos-search/issues/389). It would be nice to be more resilient to such failures, for example:

- in the particular case of maintainers, allow parsing a string as an entry `{name: name, github: null, email: null}`
- in general, don't fail the whole import if a single package or option fails to deserialise

**Comment by piegamesde** [comment]
I think we should fix this the other way around: putting a string into `meta.maintainers` should give an evaluation error in `nixpkgs`.

**Comment by sternenseemann** [comment]
This should already be inforced by ofborg in nixpkgs via `check-meta.nix` which restricts the `maintainers` field to `listOf (attrsOf str)`.

**Comment by ncfavier** [comment]
Ah but `(listOf (attrsOf str)).check [ "foo" ]` evaluates to `true`... Apparently [`listOf`](https://github.com/NixOS/nixpkgs/blob/59bfda72480496f32787cec8c557182738b1bd3f/lib/types.nix#L369) and [`attrsOf`](https://github.com/NixOS/nixpkgs/blob/59bfda72480496f32787cec8c557182738b1bd3f/lib/types.nix#L398) don't bother to check the types of their elements?

**Comment by sternenseemann** [comment]
Right, that's weird, but maybe something we can fix upstream, although I fear it may be quite the performance penalty for the already slow module system (the `Of` part seems to be mostly used to choose the proper merging behavior).

**Comment by ncfavier** [comment]
The current import failure is caused by a use of `literalDocBook` in an option `description` (https://github.com/NixOS/nixpkgs/pull/155928).

**Comment by garbas** [comment]
I think we should look again how are we importing flakes and nixpkgs.

How it works right now is quite easy to miss if some flakes are failing to import. With nixpkgs is a little better (we at least see a failed github action), but we are not really reporting it somewhere.

We need somehow to put things that fail into import into quarantine and report them to our matrix channel. And for things that are in quarantine we should use data from last import.
